lib/tests/local_request_private_cookie-issue-368.rs

```diff
@@ -0,0 +1,30 @@
+#![feature(plugin, custom_derive, decl_macro)]
+#![plugin(rocket_codegen)]
+
+extern crate rocket;
+
+use rocket::http::{Cookie, Cookies};
+
+#[get("/")]
+fn return_private_cookie(mut cookies: Cookies) -> Option<String> {
+    match cookies.get_private("cookie_name") {
+        Some(cookie) => Some(cookie.value().into()),
+        None => None,
+    }
+}
+
+mod tests {
+    use super::*;
+    use rocket::local::Client;
+
+    #[test]
+    fn private_cookie_works() {
+        let rocket = rocket::ignite().mount("/", routes![return_private_cookie]);
+
+        let client = Client::new(rocket).unwrap();
+        let req = client.get("/").private_cookie(Cookie::new("cookie_name", "cookie_value"));

```

Comment:
Can we also add a test that adds a _regular_ cookie named `cookie_name` and ensures we don't get "cookie_value" as a value?